wsmaster/che-core-api-machine/src/main/java/org/eclipse/che/api/machine/server/model/impl/MachineSourceImpl.java

```diff
@@ -25,15 +27,28 @@
 
     private String type;
     private String location;
+    private String content;
+
+    protected MachineSourceImpl() {
+
+    }
 
     public MachineSourceImpl(String type, String location) {
+        this(type, location, null);
+    }
+
+    public MachineSourceImpl(String type, String location, String content) {
```

Comment:
the problem is how do you differentiate MachineSourceImpl(String type, String location) from MachineSourceImpl(String type, String content)

I've removed the location or content from constructor.